### PR TITLE
[pkg][fix] - Filter non-fatal errors and update go.mod

### PR DESCRIPTION
**Pull request by VihasMakwana**
## What does this PR do?

With https://github.com/elastic/elastic-agent-system-metrics/issues/164, we now also expect a list of non-fatal errors (access denied, operation not permitted, invalid parameters etc).
Such errors are safe to ignore, so update the tests
Also,

- update go.mod

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

- See https://github.com/elastic/elastic-agent-system-metrics/pull/172 for information around non-fatal errros.

### CODE REVIEW

**Comment by michalpristas** [comment]
seems like notice in not properly updated, can you run `go mod tidy;mage fmt;mage update;mage notice` again

File: go.mod [hunk 1]
```diff
@@ -14,7 +14,7 @@ require (
 	github.com/elastic/elastic-agent-autodiscover v0.8.0
 	github.com/elastic/elastic-agent-client/v7 v7.15.0
 	github.com/elastic/elastic-agent-libs v0.9.15
-	github.com/elastic/elastic-agent-system-metrics v0.10.4
+	github.com/elastic/elastic-agent-system-metrics v0.10.6-0.20240723152150-0f81a1e67528
```
Comment (AndersonQ) [inline go.mod:17]: Please release `elastic-agent-system-metrics` and use the proper version

**Review by AndersonQ — COMMENTED** [review]
once you fix the `elastic-agent-system-metrics` it is good to go

**Review by blakerouse — APPROVED** [review]
Looks good to me.

**Comment by pierrehilbert** [comment]
@VihasMakwana shouldn't we backport this on to 8.15?

**Pull request merged**